### PR TITLE
Fix for CW: Items weren't cached (alt)

### PR DESCRIPTION
**Pull request by MrPetovan**
Alternative to #4597

### CODE REVIEW

**Comment by annando** [comment]
Problem is that in ```put_item_in_cache``` the item array is called by reference. So a changed body element would be stored in the item table. For this reason we are saving the old body some lines below.

**Comment by MrPetovan** [comment]
*sigh*

I've guarded against the side-effects but we really should drop the by-reference parameters both from `put_item_in_cache()` and `redir_private_images()`

File: include/text.php [hunk 1]
```diff
@@ -1169,36 +1169,45 @@ function redir_private_images($a, &$item)
...
 
+/**
+ * Sets the "rendered-html" field of the provided item
+ *
+ * Body is preserved to avoid side-effects as we modify it just-in-time for spoilers and private image links
+ *
+ * @param array $item
+ * @param bool  $update
+ *
+ * @todo Remove reference, simply return "rendered-html"
```
Comment (annando) [inline include/text.php]: We need both "rendered-html" and "rendered-hash" in return.

Comment (annando) [reply include/text.php]: Obviously this functionality needs some severe reworking :-) It was implanted some time when I hadn't got the knowledge about the system like I'm having it now.

**Pull request merged**